pkg/agent/openflow/client.go

```diff
@@ -157,6 +157,25 @@ type Client interface {
 	// SNAT with the Openflow NAT action.
 	InstallExternalFlows() error
 
+	// InstallSNATMarkFlows installs flows for a local SNAT IP. On Linux, a
+	// single flow is added to mark the packets tunnelled from remote Nodes
+	// that should be SNAT'd with the SNAT IP. On Windows, an extra flow is
+	// added to perform SNAT for the marked packets with the SNAT IP.
+	InstallSNATMarkFlows(snatIP net.IP, mark uint32) error
+
+	// UninstallSNATMarkFlows removes the flows installed to set the packet
+	// mark for a SNAT IP.
+	UninstallSNATMarkFlows(mark uint32) error
+
+	// InstallSNATPolicyFlow installs the SNAT policy flows for a local Pod.
+	// which set the SNAT IP mark on the packets from the ofPort to external.
+	// As of now, a Pod can be configured with a single SNAT IP in a single
+	// address family (IPv4 or IPv6).
+	InstallPodSNATFlows(ofPort uint32, snatMark uint32, isIPv6 bool) error
```

Comment:
@jianjuns should there be another interface to forward traffic of Pods whose SNAT IPs are not local to tunnel port with specific tun dest IP specified?
cc @ceclinux

Reply:
You are right! I missed the flow. Added it, but still need to fix the integration test failure.